server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MailboxFactory.java

```diff
@@ -106,8 +112,10 @@ public MailboxBuilder builder() {
         return new MailboxBuilder(this);
     }
 
-    private Mailbox fromMessageManager(MessageManager messageManager, Optional<List<MailboxMetaData>> userMailboxesMetadata,
-                                                 MailboxSession mailboxSession) throws MailboxException {
+    private Mailbox fromMessageManager(MessageManager messageManager,
+                                       Optional<List<MailboxMetaData>> userMailboxesMetadata,
+                                       Optional<Quotas> preloadedDefaultUserQuotas,
```

Comment:
I don't agree to make MailboxFactory more complex to handle a special case of a field being set with some preloaded values.
We could for example extract quota loading logic in a component and we take it there, opening the door to various implementations of that loader.
Or we can find any better design we want, but adding `if`s in MailboxFactory is not an option IMO.